src/v/datalake/batching_parquet_writer.cc

```diff
@@ -159,4 +162,26 @@ ss::future<> batching_parquet_writer::abort() {
     }
 }
 
+batching_parquet_writer_factory::batching_parquet_writer_factory(
+  std::filesystem::path local_directory,
+  size_t row_count_threshold,
+  size_t byte_count_threshold)
+  : _local_directory{std::move(local_directory)}
+  , _row_count_threshold{row_count_threshold}
+  , _byte_count_treshold{byte_count_threshold} {}
+
+ss::future<std::unique_ptr<data_writer>>
+batching_parquet_writer_factory::create_writer(iceberg::struct_type schema) {
+    auto ret = std::make_unique<batching_parquet_writer>(
+      std::move(schema), _row_count_threshold, _byte_count_treshold);
+    std::string filename = fmt::format("{}.parquet", uuid_t::create());
+    std::filesystem::path file_path = _local_directory / filename;
+    auto err = co_await ret->initialize(file_path);
+    if (err != data_writer_error::ok) {
+        // FIXME: This method should return a result and let the multiplexer
+        // deal with it appropriately
```

Comment:
We should probably have thrown at this level instead of the next layer up.

Reply:
I'm not sure what you mean by "this level" you mean throw an exception up to the multiplexer?